src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -330,8 +332,38 @@ public abstract class GenerateProtoTask extends DefaultTask {
   @Internal("Input captured by getAlternativePaths(), this is used to query alternative path by locator name.")
   abstract MapProperty<String, FileCollection> getLocatorToAlternativePathsMapping()
 
-  @InputFiles
-  @PathSensitive(PathSensitivity.NONE)
+  @Input
+  @Optional
+  Provider<String> getProtocArtifact() {
+    // Ideally, this value could be mapped from the protocLocator property, but Provider#map cannot return null in
+    // Gradle < 6.2. See https://github.com/gradle/gradle/issues/11979
+    return providerFactory.provider {
+      def protocArtifact = protocLocator.get().artifact
+      if (protocArtifact) {
+        def (groupId, artifact, version) = ToolsLocator.artifactParts(protocArtifact)
+        outputs.doNotCacheIf("protoc is a SNAPSHOT version", {
```

Comment:
It seems like maybe you could try to go cache based off of the SNAPSHOT file itself, but it seemed like it would complicate things and not provide much value for our customers or typical non-Protobuf devs.